plugins/packages/woocommerce/lib/index.ts

```diff
@@ -0,0 +1,285 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-plugins/common';
+import { SourceOptions, QueryOptions } from './types';
+const JSON5 = require('json5');
+
+export default class Woocommerce implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions): Promise<QueryResult> {
+    const operation = queryOptions.operation;
+    let result = {};
+    const { product_id, order_id, customer_id, body } = queryOptions;
+    const WooCommerce = await this.getConnection(sourceOptions);
+
+    try {
+      switch (operation) {
+        case 'list_customer': {
+          result = await WooCommerce.get('customers')
+            .then((response) => {
+              return response?.data;
+            })
+            .catch((error) => {
+              return error.response.data;
```

Comment:
How does the user know if its success or failure, shall we add status code also to result? both for success and failure cases

Reply:
error already have status returned
<img width="634" alt="Screenshot 2022-04-21 at 12 02 52 PM" src="https://user-images.githubusercontent.com/31006028/164389978-17bf3ece-3639-4b78-ac1e-0070123c4c6e.png">

for successful  operations i am attaching statucode with response
<img width="1280" alt="Screenshot 2022-04-20 at 9 25 42 AM" src="https://user-images.githubusercontent.com/31006028/164390196-fce2160b-caa5-4c20-8d11-ad4feeb398e8.png">